src/components/Navigation/index.tsx

```diff
@@ -0,0 +1,10 @@
+"use client";
+import useMediaQuery from "@/hooks/useMediaQuery";
+import DesktopNavigation from "./DesktopNavigation";
+import MobileNavigation from "./MobileNavigation";
+
+export default function Navigation() {
+  const isDesktop = useMediaQuery("(min-width: 768px)");
+
+  return <>{isDesktop ? <DesktopNavigation /> : <MobileNavigation />}</>;
```

Comment:
<></> 구문 필요없이 early return으로 작성하면 더 간결할 것 같아요~

구간이  늘어날때 대응도 쉬울 수  있고요

Reply:
아 리액트에서는 컴포넌트도 early return이 되는군요??? 이런 느낌일까요??
```suggestion
  if (isDesktop) {
    return <DesktopNavigation />;
  }
  
  return <MobileNavigation />;
```

그런데 저는 사실 early return은 if else 문에서 에러 혹은 예외 상황 발생 케이스를 처리 후 먼저 반환해서 가독성을 높일 때 사용하는 방식이라 생각해서요. 예를 들어 다음과 같은 경우요
```js
function handler() {
  if (isSomethingEmpty) {
    // 에러처리
    return
  }

  // 정상일 때 처리
}
```
사실 화면 너비가 데스크탑인지의 여부는 에러라기보다는 말 그대로 두 가지 케이스 중에 하나여서, 오히려 early return보다는 if else문이나 삼항 연산자로 처리하는게 더 명확할 것 같다는 생각이 듭니다. 어떻게 생각하시나요?

Reply:
조건이 지금처럼 한개이면 `isDesktop ? <DesktopNavigation /> : <MobileNavigation />` 로 해도  상관이 없는데 나중에  break point가 생기면 if 구문이 더 편할  수  있습니다.

리액트에서 상태가 많아지면 삼항연산자가 복잡해져서 early return으로 흔히 사용합니다. early return으로 변수의 타입도  제한이 되어서  많이들 사용하긴 합니다.

early return도 길어지면 결국 복잡해져서 ts-pattern 같은 라이브러리를 사용하기도  합니다.

그런데  ts-pattern도 한계점이 있어서 굳이 사용하지 않고 있긴합니다.

https://toss.tech/article/ts-pattern-usage

지금  경우에는 <></>만 없애도 좋을것 같네요

Reply:
넵, 분기가 늘어나는 것은 나중에 정말로 조건이 더 늘어났을 때 고민해도 괜찮을 것 같아요.
리액트에서는 if else문보다 early return을 사용해서 컴포넌트 분기 처리를 많이 하나 보네요. 확실히 코드가 더 깔끔해진다는 장점이 있으니... 일단 빈 <> 태그부터 없애고, 추후에 분기문이 더 추가된다면 early return 사용하도록 하겠습니다!